dimod/response.py

```diff
@@ -13,737 +13,469 @@
-            NumPy array-like object.
+            sorted_by (str/None, optional, default='energy'):
+                Selects the record field used to sort the samples. If None, the samples are yielded
+                in record order.
 
-        Raises:
-            :exc:`ValueError`: If vartype is not provided and samples are all 1s, have more
-                than two unique values, or have values of an unknown vartype.
+        Yields:
+            :obj:`.SampleView`: A view object mapping the variable labels to their values. Acts like
+            a read-only dict.
 
         Examples:
-            This example code snippet builds a response from a NumPy matrix.
-
-            .. code-block:: python
-
-                samples = np.matrix([[0, 1], [1, 0]])
-                energies = [0.0, 1.0]
-                response = Response.from_matrix(samples, {'energy': energies})
-
-            This example code snippet builds a response from a NumPy array-like object (a Python list).
-
-            .. code-block:: python
-
-                samples = [[0, 1], [1, 0]]
-                energies = [0.0, 1.0]
-                response = Response.from_matrix(samples, {'energy': energies})
 
-        Notes:
-            SciPy defines array_like in the following way: "In general, numerical data arranged in
-            an array-like structure in Python can be converted to arrays through the use of the
-            array() function. The most obvious examples are lists and tuples. See the documentation
-            for array() for details for its use. Some objects may support the array-protocol and
-            allow conversion to arrays this way. A simple way to find out if the object can be
-            converted to a numpy array using array() is simply to try it interactively and see if it
-            works! (The Python Way)." [array_like]_
-
-        References:
-            .. [array_like] Docs.scipy.org. (2018). Array creation - NumPy v1.14 Manual. [online]
-                Available at: https://docs.scipy.org/doc/numpy/user/basics.creation.html
-                [Accessed 16 Feb. 2018].
+            >>> import dimod
+            ...
+            >>> response = dimod.ExactSolver().sample_ising({'a': 0.0, 'b': 0.0}, {('a', 'b'): -1})
+            >>> for sample in response.samples():   # doctest: +SKIP
+            ...     print(sample)
+            {'a': -1, 'b': -1}
+            {'a': 1, 'b': 1}
+            {'a': 1, 'b': -1}
+            {'a': -1, 'b': 1}
 
         """
-        samples_matrix = np.matrix(samples, dtype=np.int8)
-
-        if vartype is None:
-            vartype = infer_vartype(samples_matrix)
-
-        response = cls(samples_matrix, data_vectors=data_vectors,
-                       vartype=vartype, info=info, variable_labels=variable_labels)
-
-        return response
+        if n is None:
+            for sample, in self.data(['sample'], sorted_by=sorted_by, name=None):
```

Comment:
Syntax error? Or at least runtime error?

Reply:
I am not sure what you mean here